### PR TITLE
108 post chatroom

### PR DESCRIPTION
**Pull request by DLiSWE**
[Post] /chatroom

### CODE REVIEW

File: src/services/chatroom.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Chatroom } from '@/entities/nosql';
+import { HttpException } from '@/exceptions';
+import { model } from 'mongoose';
+
+class ChatroomService {
+  public async createChatroom(userId: number, receiverId: number): Promise<void> {
+    const chatroom = model('Chatroom', Chatroom);
+    const createChat = new chatroom({
+      isGroup: true,
```
Comment (Matt-Tan15) [inline src/services/chatroom.service.ts:9]: should this always be true? if it's just a one to one chatroom it's not really a group

**Review by emir4ik-code — DISMISSED** [review]

File: src/services/chatroom.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class ChatroomService {
+  public async createChatroom({ userId, receiverId, isGroup }: CreateChatroomDto): Promise<void> {
+    const chatroom = model('Chatroom', Chatroom);
+    const createChat = new chatroom({
+      isGroup: isGroup,
+      users: [userId, receiverId],
+      message: [],
+      createdDate: Date.now(),
+    });
+    const checkExist = await chatroom.exists({ users: createChat.users });
```
Comment (ZohaibManzoor00) [inline src/services/chatroom.service.ts:15]: Shouldn't we check for if the chatroom exists before we create one?
so line 15 and 16 would be at the top.

Comment (DLiSWE) [reply src/services/chatroom.service.ts]: True, I did create the object. But I didn't save it until after checking whether it exists. I changed it though

**Review by emir4ik-code — APPROVED** [review]

**Review by Matt-Tan15 — APPROVED** [review]
Nice

**Pull request merged**